engine/worker/run_plugin.go

```diff
@@ -123,47 +128,45 @@ func startGRPCPlugin(ctx context.Context, pluginName string, w *currentWorker, p
 	cmd := binary.Cmd
 	if _, err := exec.LookPath(cmd); err != nil {
 		cmd = path.Join(w.basedir, cmd)
-		_, err = exec.LookPath(cmd)
-		if err != nil {
-			return nil, sdk.WrapError(err, "plugin:%s unable to start GRPC plugin, binary command not found.", pluginName)
+		if _, err := exec.LookPath(cmd); err != nil {
+			return nil, fmt.Errorf("plugin:%s unable to start GRPC plugin, binary command not found (%v)", pluginName, err)
 		}
 	}
 	args := append(binary.Entrypoints, binary.Args...)
-
-	if err := grpcplugin.StartPlugin(ctx, dir, cmd, args, envs, &c.BuffOut); err != nil {
-		return nil, sdk.WrapError(err, "plugin:%s unable to start GRPC plugin... Aborting", pluginName)
+	var errstart error
+	if c.StdoutPipe, c.StderrPipe, errstart = grpcplugin.StartPlugin(ctx, dir, cmd, args, envs); errstart != nil {
+		return nil, fmt.Errorf("plugin:%s unable to start GRPC plugin... Aborting. err:%v", pluginName, errstart)
 	}
 	log.Info("GRPC Plugin %s started", binary.Name)
 
 	//Sleep a while, to let the plugin write on stdout the socket address
 	time.Sleep(500 * time.Millisecond)
 	tsStart := time.Now()
 
-	buff := new(strings.Builder)
+	stdoutreader := bufio.NewReader(c.StdoutPipe)
+
 	for {
-		b, err := c.BuffOut.ReadByte()
-		if err != nil && len(buff.String()) > 0 {
-			buff.Reset()
+		line, errs := stdoutreader.ReadString('\n')
+		if errs == io.EOF {
+			continue
+		}
+		if errs != nil {
 			if time.Now().Before(tsStart.Add(5 * time.Second)) {
-				log.Warning("plugin:%s error on ReadByte, retry in 500ms...", pluginName)
+				log.Warning("plugin:%s error on ReadString, retry in 500ms...", pluginName)
 				time.Sleep(500 * time.Millisecond)
 				continue
 			}
-			log.Error("plugin:%s error on ReadByte(len buff %d, content: %s): %v", pluginName, len(buff.String()), buff.String(), err)
-			return nil, fmt.Errorf("unable to get socket address from started binary")
-		}
-		if err := buff.WriteByte(b); err != nil {
-			log.Error("plugin:%s error on write byte: %v", pluginName, err)
-			break
+			log.Error("plugin:%s error on ReadString(len buff %d, content: %s): %v", pluginName, len(line), line, err)
+			continue
 		}
-		if strings.HasSuffix(buff.String(), "is ready to accept new connection\n") {
+		if strings.HasSuffix(line, "is ready to accept new connection\n") {
+			socket := strings.TrimSpace(strings.Replace(line, " is ready to accept new connection\n", "", 1))
```

Comment:
idea: because this line is part of the "protocol" used for plugin connection, this line should be printed and parsed by a func of the sdk.

Reply:
done